waku/waku_store_v3/client.nim

```diff
@@ -0,0 +1,210 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/options,
+  stew/results,
+  chronicles,
+  chronos,
+  metrics,
+  bearssl/rand
+import
+  ../node/peer_manager,
+  ../utils/requests,
+  ./protocol_metrics,
+  ./common,
+  ./rpc_codec
+
+when defined(waku_exp_store_resume):
+  import std/[sequtils, times]
+  import ../waku_archive
+  import ../waku_core/message/digest
+
+logScope:
+  topics = "waku store client"
+
+const DefaultPageSize*: uint = 20 # A recommended default number of waku messages per page
+
+type WakuStoreV3Client* = ref object
```

Comment:
Since the `V3` is essentially a temporary measure to disambiguate with the "legacy" store protocol, perhaps another pattern is to design type names for the new (v3) Store to be exactly like we envision them to be in the long term and disambiguate on `waku_node` by:
- using aliases for the imports related to legacy Store (e.g. `./waku_store/client as legacy_store_client`)
- renaming where necessary the _legacy_ types, metrics, etc. to `...Legacy` (e.g. `WakuStoreClientLegacy`)

This would basically prepare the codebase to remove the legacy store protocol cleanly in the future without having to change "v3" implementation again.